File: admin-angular/src/app/models/update-requests/product-update-request.ts
import { AutoMap } from "@automapper/classes";
import { BaseUpdateRequest } from "..";
import { ProductDetailUpdateRequest } from "./product-detail-update-request";

export class ProductUpdateRequest extends BaseUpdateRequest  {
    @AutoMap()
    id?: number;
    @AutoMap()
    categoryId?: number;
    @AutoMap()
    price?: number;
    @AutoMap()
    code?: string;
    @AutoMap()
    imageUrl?: string;
    @AutoMap({ typeFn: () => ProductDetailUpdateRequest })
    details: ProductDetailUpdateRequest[]
}

File: App.Models/DTOs/Products/ProductUpdateRequest.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using MaxV.Common.Model.DTOs;
using App.Models.DTOs.ProductDetails;
using App.Models.DTOs.UpdateRquests;

namespace App.Models.DTOs.Products
{
    public class ProductUpdateRequest : BaseUpdateRequest<int>
    {
        public int CategoryId { get; set; }
        public string CategoryName { get; set; }
        [MaxLength(256)]
        public string Code { get; set; }
        public double? Price { get; set; }
        [Range(0, 100)]
        public int? PercentDiscount { get; set; }
        public double? MaxDiscountPrice { get; set; }
        public string ImageUrl { get; set; }
        [JsonProperty("file")]
        public IFormFile File { get; set; }
        [JsonProperty("details")]
        [FromForm(Name = "details")]
        public virtual List<ProductDetailUpdateRequest> ProductDetails { get; set; }
        public string Name { get; set; }
    }
}
